project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc

```diff
@@ -179,6 +179,42 @@ function c4m_user_notifications_content_recommended($flag, $entity_id, $account,
  * Email notification about new content at organic group.
  */
 function c4m_user_notifications_new_group_content_published($node) {
+  _c4m_user_notifications_new_content_published($node);
+}
+
+/**
+ * Helper function.
+ *
+ * Sends mail notification to group subscribers, when new content is added.
+ *
+ * @param object $node
+ *   Content node that was added.
+ */
+function _c4m_user_notifications_new_content_published($node) {
+  global $user;
+  $message = message_create('c4m_new_content_published');
+  c4m_message_load_tokens__c4m_new_content_published($message);
+  $body_teaser = field_view_field('node', $node, 'c4m_body', 'teaser');
+  $message->arguments['@node_teaser_body'] = strip_tags($body_teaser[0]['#markup']);
+
+  $wrapper = entity_metadata_wrapper('message', $message);
```

Comment:
@anvmn 
a lot of stuff are happening in here, please add elaborated comments, thanks.